packages/browser-lite/src/database.ts

```diff
@@ -0,0 +1,552 @@
+// tslint:disable:max-file-line-count
+
+import Dexie from 'dexie';
+
+export type Record = Order | MiniHeader | Metadata;
+
+export interface Options {
+    dataSourceName: string;
+    maxOrders: number;
+    maxMiniHeaders: number;
+}
+
+export interface Query<T extends Record> {
+    filters?: Array<FilterOption<T>>;
+    sort?: Array<SortOption<T>>;
+    limit?: number;
+    offset?: number;
+}
+
+export interface SortOption<T extends Record> {
+    field: Extract<keyof T, string>;
+    direction: SortDirection;
+}
+
+export interface FilterOption<T extends Record> {
+    field: Extract<keyof T, string>;
+    kind: FilterKind;
+    value: any;
+}
+
+export enum SortDirection {
+    Asc = 'ASC',
+    Desc = 'DESC',
+}
+
+export enum FilterKind {
+    Equal = '=',
+    NotEqual = '!=',
+    Less = '<',
+    Greater = '>',
+    LessOrEqual = '<=',
+    GreaterOrEqual = '>=',
+    Contains = 'CONTAINS',
+}
+
+export interface Order {
```

Comment:
`"common/types".OrderWithMetdata` is named that way for a very specific reason. We have lots of different order types in that package that are used in different contexts: `Order`, `SignedOrder`, `OrderInfo`, etc. We needed a name that distinguishes it from these other types.

In most other places where only one order type is used for the package (e.g. `sqltypes`, `dexietypes`), we just call it `Order`. There is no risk of confusion because of how types are referenced across packages in Go.

Honestly, I think there is also little chance of confusion in TypeScript land either, but I would be willing to rename it. How do you feel about `DexieOrder` or `DBOrder` since this is specifically for interfacing with Dexie.js in this context?

Reply:
I'm okay with changing it. After I reviewed this file, I saw that `Order` was used directly in `db/dexie_implementation.go` and `db/sql_implementation.go`.